packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -3,14 +3,87 @@
 // found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:developer';
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
 
 import 'binary_messenger.dart';
 import 'binding.dart';
 import 'message_codec.dart';
 import 'message_codecs.dart';
 
+bool debugProfilePlatformChannels = true;
+bool _debugProfilePlatformChannelsIsRunning = false;
+
+class _PlatformChannelStats {
+  _PlatformChannelStats(this.channel, this.codec, this.type);
+
+  final String channel;
+  final String codec;
+  final String type;
+
+  int _upCount = 0;
+  int _upBytes = 0;
+  int get upBytes => _upBytes;
+  void addUpStream(int bytes) {
+    _upCount += 1;
+    _upBytes += bytes;
+  }
+
+  int _downCount = 0;
+  int _downBytes = 0;
+  int get downBytes => _downBytes;
+  void addDownStream(int bytes) {
+    _downCount += 1;
+    _downBytes += bytes;
+  }
+
+  double get averageUpPayload => _upBytes / _upCount;
+  double get averageDownPayload => _downBytes / _downCount;
+}
+
+final Map<String, _PlatformChannelStats> _debugProfilePlatformChannelsStats = <String, _PlatformChannelStats>{};
+
+void _debugLaunchProfilePlatformChannels() async {
+  if (!_debugProfilePlatformChannelsIsRunning) {
+    _debugProfilePlatformChannelsIsRunning = true;
+    await Future<dynamic>.delayed(const Duration(seconds: 1));
+    _debugProfilePlatformChannelsIsRunning = false;
+    String log = 'Platform Channel Stats:\n';
+    final List<_PlatformChannelStats> allStats =
+        _debugProfilePlatformChannelsStats.values.toList();
+    allStats.sort((_PlatformChannelStats x, _PlatformChannelStats y) =>
+        (y.upBytes + y.downBytes) - (x.upBytes + x.downBytes));
+    for (final _PlatformChannelStats stats in allStats) {
+      log +=
+          '  (name:${stats.channel} type:${stats.type} codec:${stats.codec} up:${stats.upBytes} up_avg:${stats.averageUpPayload} down:${stats.downBytes} down_avg:${stats.averageDownPayload})\n';
+    }
+    print(log);
+    _debugProfilePlatformChannelsStats.clear();
+  }
+}
+
+void _debugRecordUpStream(
+    Object channel, String name, Object codec, ByteData? bytes) {
+  final _PlatformChannelStats stats =
+      _debugProfilePlatformChannelsStats[name] ??=
+          _PlatformChannelStats(
+              name, codec.toString(), channel.runtimeType.toString());
```

Comment:
runtimeType.toString is expensive.

codec.toString may be expensive.

Reply:
These are only run once per channel per process so it should be fine.